### PR TITLE
Add basic tests for folder REST update

### LINKED ISSUE #63: Implement remaining Folder REST tests
Still need `test_folder_rest_update` and `test_folder_rest_destroy`. Both might need to be split into multiple cases, if we want to test behaviors for root / child or other varying circumstances.

* https://github.com/girder/dkc-next/blob/master/dkc/core/tests/test_folder_rest.py#L52
* https://github.com/girder/dkc-next/blob/master/dkc/core/tests/test_folder_rest.py#L58

### PR DESCRIPTION
**Pull request by zachmullen**
Fixes #63

### CODE REVIEW

**Comment by zachmullen** [comment]
@brianhelba PTAL, this should be good to merge

File: dkc/core/tests/test_folder_rest.py [hunk 1]
```diff
@@ -47,11 +47,24 @@ def test_folder_rest_retrieve(api_client, folder):
...
+    folder.refresh_from_db()
+    assert folder.name == 'New name'
+    assert folder.description == 'New description'
+
+
+@pytest.mark.django_db
+def test_folder_rest_update_parent_disallowed(api_client, folder, folder_factory):
+    other_root = folder_factory()
+    child = folder_factory(parent=folder)
+    api_client.patch(f'/api/v2/folders/{child.id}', data={'parent': other_root.id})
```
Comment (jbeezley) [inline dkc/core/tests/test_folder_rest.py:65]: I would assert the status code returned is correct.

**Review by jbeezley — CHANGES_REQUESTED** [review]

File: dkc/core/tests/test_folder_rest.py [hunk 2]
```diff
@@ -62,7 +62,8 @@ def test_folder_rest_update(api_client, folder):
 def test_folder_rest_update_parent_disallowed(api_client, folder, folder_factory):
     other_root = folder_factory()
     child = folder_factory(parent=folder)
-    api_client.patch(f'/api/v2/folders/{child.id}', data={'parent': other_root.id})
+    resp = api_client.patch(f'/api/v2/folders/{child.id}', data={'parent': other_root.id})
+    assert resp.status_code == 200
```
Comment (jbeezley) [inline dkc/core/tests/test_folder_rest.py:66]: I would expect that to be an error.  Is 200 the behavior dictated by DRF?

Comment (zachmullen) [reply dkc/core/tests/test_folder_rest.py:66]: It is indeed DRF's behavior when trying to patch a non-editable field. It ignores it and continues on. I also would have hoped for a 400 here.

Comment (jbeezley) [reply dkc/core/tests/test_folder_rest.py:66]: Apparently this is the default behavior of DRF when given an unknown field.  I wonder if making parent a [read_only_field](https://www.django-rest-framework.org/api-guide/serializers/#specifying-read-only-fields) would raise an error?

Comment (jbeezley) [reply dkc/core/tests/test_folder_rest.py:66]: I guess it doesn't... oh well, I guess this is good enough.

**Review by jbeezley — APPROVED** [review]

**Pull request merged**